File: lab2/pi_cal.f90
program pi_cal
  use omp_lib
  implicit none
  integer :: i, niter
  real(8) :: pi, start, end

  ! Initialize variables
  pi = 0.0d0
  niter = 1000000000

  ! Timing
  start = omp_get_wtime()

  ! Calculate part of Leibnitz sum in parallel
  !$omp parallel do reduction(+:pi)
  do i = 0, niter-1
     pi = pi + real((-1)**i, 8) * (4.0d0 / (2.0d0*real(i, 8)+1.0d0))
  end do
  !$omp end parallel do

  ! Stop timing
  end = omp_get_wtime()

  ! Print result
  write(*, '(a, f0.10, a, f6.3, a)', advance='no') "pi estimate is ", pi, ", obtained in ", end-start, " seconds in Fortran"
  write(*, *)
end program pi_cal

File: lab2/pi_cal.c
#include <omp.h>
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

int main (int argc, char *argv[])
{

  // Initialize variables
  int i;
  double pi = 0;
  int niter = 1000000000;

  // Timing
  double start,end;
  start=omp_get_wtime();

  // Calculate part of Leibnitz sum.
  #pragma omp parallel for reduction(+:pi)
  for(i = 0; i < niter; i++)
  {
    pi = pi + pow(-1, i) * (4 / (2*((double) i)+1));
  }

  // Stop timing
  end=omp_get_wtime();

  // Print result
  printf("pi estimate is %f, obtained in %f seconds in C\n", pi, end-start);

}
